javalib/src/main/scala/java/lang/Runtime.scala

```diff
@@ -18,6 +19,17 @@ class Runtime private () {
   lazy val setupAtExitHandler = {
     stdlib.atexit(() => Runtime.getRuntime().runHooks())
   }
+
+  lazy val setupSignalHandler = {
+    signal.signal(signal.SIGINT, handleSignal(_))
+    signal.signal(signal.SIGTERM, handleSignal(_))
+  }
```

Comment:
```suggestion
  lazy val setupSignalHandler = {
    signal.signal(signal.SIGHUP, handleSignal(_))
    signal.signal(signal.SIGINT, handleSignal(_))
    signal.signal(signal.SIGQUIT, handleSignal(_))
    signal.signal(signal.SIGTERM, handleSignal(_))
  }
```
let's add a few more expected reason to quit

Reply:
As far as I understand "The JVM runs shutdown hooks only in case of normal terminations." - what this seems to mean is Ctrl-C `SIGINT` and `SIGTERM`. This is also in the current Test comments.

Reply:
> The JVM catches signals to implement shutdown hooks for unexpected termination. The JVM uses SIGHUP, SIGINT, and SIGTERM to initiate the running of shutdown hooks.
> Applications embedding the JVM frequently need to trap signals such as SIGINT or SIGTERM, which can lead to interference with the JVM signal handlers. The -Xrs option is available to address this issue. When -Xrs is used, the signal masks for SIGINT, SIGTERM, SIGHUP, and SIGQUIT aren't changed by the JVM, and signal handlers for these signals aren't installed.

https://docs.oracle.com/en/java/javase/21/docs/specs/man/java.html

Reply:
Ok, I'll add `SIGHUP`.

Reply:
On second thought, we need to try and get this to work properly this way first as we are using `clib` and `SIGHUP` is not a member. I second pass could use `posixlib` and something special for Windows to make it JVM compliant.

Reply:
Keep in mind that `SIGHUP` is a signal which is send when SSH connection is hand up due network issue.

Without it, shutdown hook at CLI utilite won't work on this case and it's show stoper for real life use case I guess.

Reply:
No problem but we have nothing now and to make it cross platform is more complex so it might be better to have something now.